flang/docs/DoConcurrentConversionToOpenMP.md

```diff
@@ -0,0 +1,240 @@
+
+> § 3.35
+> ------
+> construct entity
+> entity whose identifier has the scope of a construct
+
+> § 19.4
+> ------
+>  A variable that appears as an index-name in a FORALL or DO CONCURRENT construct, or ... is a construct entity. A variable that has LOCAL or LOCAL_INIT locality in a DO CONCURRENT construct is a construct entity.
+> ...
+> The name of a variable that appears as an index-name in a DO CONCURRENT construct, FORALL statement, or FORALL construct has a scope of the statement or construct. A variable that has LOCAL or LOCAL_INIT locality in a DO CONCURRENT construct has the scope of that construct.
+
+From the above quotes, it seems there is an equivalence between the IV of a `do concurrent` loop and a variable with a `LOCAL` locality specifier (equivalent to OpenMP's `private` clause). Which means that we should probably localize/privatize a `do concurernt` loop's IV even if it is not perfectly nested in the nest we are parallelizing. For now, however, we **do not** that as pointed out previously. In the near future, we propose a middle-ground solution (see the Next steps section for more details).
+
+## Next steps
+
+### Delayed privatization
+
+So far, we emit the privatization logic for IVs inline in the paralle/target region. This is enough for out purposes right now since we don't localize/privatize any sophisticated types of variables yet. Once we have need for more advanced localization thought `do concurrent`'s locality specified (see below), delayed privatization will enable us to have a much cleaner IR. Once delayed privatization's implementation upstream is supported for the required constructs by the pass, we will move to it rather than inlined/early privatization.
+
+### Locality specifiers for `do concurrent`
+
+Locality specified will enable the user to control the data environment of the loop nest in a more fine-grained way. Implementing these specified on the `FIR` dialect level is needed in order to support this in the `DoConcurrentConversionPass`.
+
+Such specified will also unlock a potential solution to the non-perfectly-nested loops' IVs issue described above. In particular, for a non-perfectly nested loop, one middle-ground proposal/solution would be to:
+* Emit the loop's IV as shared/mapped just like we do currently.
+* Emit a warning that the IV of the loop is emitted as shared/mapped.
+* Given support for `LOCAL`, we can recommend the user to explicitly localize/privatize the loop's  IV if they choose to.
+
+### More advanced detection of loop nests
+
+As pointed out earlier, any intervening code between the headers of 2 nested `do concurrent` loops prevents us currently from detecting this as a loop nest. In some cases this is overly conservative. Therefore, a more flexible detection logic of loop nests needs to be implemented.
```

Comment:
Do you have already any plan in mind to decide where to place the operations that were originally located between the outer and inner loop?

Reply:
Nothing concrete yet. Stay tuned 😛.